bndtools/bndtools.workspace.min
# Update bndtools.workspace.min from OSGi R7 to R8

This relates to #5632 and #5864

As discussed in the call today this is a first attempt to update the template from R7 to R8.

## What is this for?

This template should make it easier "Getting started" with bndtools and is basically the minimum a bndtools Eclipse workspace should look like when starting a new project. 
Before this change it was based on OSGi R7.
For R8 the repositories need to be updated. It isn't as easy as changing a 7 to an 8 unfortunately.

## Current solution

I made use of the bnd command line command, which automatically updates all versions of an _.mvn_ file to the latest version.

`bnd mbr update` (which is a gem I will add to the [docu](https://bnd.bndtools.org/chapters/400-commands.html) soon. Thanks @pkriens for telling me)

## How  to test this?

This is the place where I need support to verify if we are satisfied with result, or what needs to be changed / added.

- Go to Eclipse Preferences / BndTools / Workspace Template
- Remove the Github repositories (just for testing)
<img width="1020" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/1cff8d31-d31d-46a1-a0ae-7f1a1d5822aa">

- Add this current repo  to "Raw Git Clone URLs" (`https://github.com/chrisrueger/bndtools.workspace.min.git` and branch `origin/R7-to-R8`)
<img width="586" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/f4517b4f-bcb6-47bf-84c6-523273b0f662">

- Click Apply and Close
- Go to File / New / Bnd OSGi Workspace
<img width="564" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/39878671-3f12-4098-bad7-8d77ce64e691">

- Choose new Workspace or update the current one and Click next
<img width="1184" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/48156f0b-80bd-4372-9afe-c9d43a580d46">

- Select the template from the repo you just entered above
<img width="1184" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/911386a5-420b-4a06-8085-93b886ef8341">

<img width="1184" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/d3c5df54-a7b3-4b9c-aba0-3c39ef1ad069">

- when asked to Switch to the Bnd Perpective say "Yes" or manually change there.

This should get you a BnWorkspace which looks lke this:

<img width="1728" alt="image" src="https://github.com/bndtools/bndtools.workspace.min/assets/188422/56b32ce6-e4b7-4228-98e0-4f76647c3d44">

- only the `cnf` folder
- 2 Repositories
 - OSGi R8 - Build (which contains the API bundles... probably should name it API)
 - OSGi R8 - Runtime **Distro** (a selection of implementation bundles)

The distro term is something I learned from @pkriens today which is just a collection of _implementation_ bundles.
The version in there are just based on the existing `cnf/runtime.mvn` and applying `bnd mbr update`

## Help wanted

I am not sure if the current status is ok or not. Eclipse Problems view is empty, but I need someone to verify if it is enough.
